plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -1583,6 +1583,19 @@ private void testCreatePartitionedTable(Session session, HiveStorageFormat stora
         assertFalse(getQueryRunner().tableExists(session, "test_partitioned_table"));
     }
 
+    @Test
+    public void createTableLikeIncludingPropertiesExternalTable()
```

Comment:
Please do add a test to showcase that the `create table like` statement will fail when not specifying the 
`external_location` for an `external` table.
Maybe (to emphasise better this difference) it would be worth to showcase as well in the test that the managed table creation will succeed in the case of not specifying the `external_location` property.

Similar test for iceberg/delta would be welcome

Reply:
The managed table create table like behavior is already tested in the `BaseConnectorTest` provided tests. The external tables test case is a good one to add though.

Reply:
Current behavior is that if you don't provide an external location, the table becomes a managed table and depends on the location either at schema or database level in Hive Metastore. So currently there are no failures because of missing the external_location property.

The metastore implementation in Hive, Delta and Iceberg either comes from hadoop or the `FileHiveMetastore`,

We could enhance the `FileHiveMetastore` to support a Hive metastore without warehouse directory, which is a valid setup. WDYT?

Reply:
Let's keep this PR focused on the issue at hand to avoid it spanning longer than it should.